lib/rex/ui/text/dispatcher_shell.rb

```diff
@@ -210,6 +210,31 @@ def cmd_help(cmd=nil, *ignored)
           print(File.read(File.join(docs_dir + '.md')))
         end
       end
+
+      # smb, mysql, mssql and postgresql session types make use of `shell.session` which type can then
+      # be called against (`shell.session.type`)
+      #
+      # Meterpreter seems to make use of `shell.client` which type can then be called against (`shell.client.type`)
+      #
+      # checking this to not raise a no methods error
+      unless defined?(shell.session).nil?
+        # List of session types we want to list compatible modules for.
+        # Wont include Meterpreter and shell as that would be too long a list.
+        session_types_to_display_compatible_modules_for = %w[smb mysql mssql postgresql]
```

Comment:
This breaks the open closed principal -  https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle

Is it possible to have this logic in the postgres/smb/etc session types instead? 👀 

i.e. maybe overriding the base `help_to_s ` implementation in some way for the dispatchers that want this information to appear

Reply:
I'll take a look into this 👍

Reply:
Made some changes to try and address this. Let me know if that is heading the correct direction 👍

Reply:
I tested it against all session types and it no longer interacts or need to check for Meterpreter and only comes into play when interacting with the new sessions types.